### PR TITLE
Communication between cli and daemon

### LINKED ISSUE #21: Communication between daemon local and cli

### PR DESCRIPTION
**Pull request by JeremyLARDENOIS**
Close #21

### CODE REVIEW

File: client/src/agent/list.rs [hunk 1]
```diff
@@ -9,6 +11,28 @@ impl Handler for ListCommand {
     fn handler(&self) -> Result<()> {
         log::info!("This should print the files");
```
Comment (AlexandreBrg) [inline client/src/agent/list.rs:12]: Please remove this line, it's not useful

File: client/src/main.rs [hunk 2]
```diff
@@ -119,6 +127,38 @@ async fn main() -> Result<()> {
...
+            for stream in listener.incoming().filter_map(handle_error) {
+                println!("Incoming connection!, {:?}", stream);
+                let mut reader = BufReader::new(stream);
+                // read in vec
+                // let mut buffer = Vec::new();
+                // reader.read_to_end(&mut buffer).expect("failed to read");
+                // let command = String::from_utf8(buffer)
+                //     .expect("failed to convert buffer to string")
+                //     .trim()
+                //     .to_string();
```
Comment (AlexandreBrg) [inline client/src/main.rs:142]: Please remove, not useful

**Review by AlexandreBrg — CHANGES_REQUESTED** [review]
Nice work, working great on Ubuntu

**Comment by AlexandreBrg** [comment]
* Please handle the deletion of `polydrive.sock` file in order to have gracefull shutdown.  SIGKILL + SIGINT 
* Make `cargo test` run onto `ubuntu 18/21` and `macOs 11`, see [github-env](https://github.com/actions/virtual-environments) for more infos 
* Add a test in `main.rs` to test whether the socket is working properly

**Comment by AlexandreBrg** [comment]
Hi @JeremyLARDENOIS, @thomasgouveia finished your PR for the cli & communication with sockets. Your help was precious for his work. However, can you now focus yourself on testing if sockets are working properly in tests ? Even though we talked that we might not need it, it should be a good exercie for you to do that. 

There are [macros](https://doc.rust-lang.org/core/arch/index.html) to test for each platform you run tests on. Use them in order to have tests on this feature.

**Comment by JeremyLARDENOIS** [comment]
Ubuntu 21 doesn't exists in github actions https://github.com/actions/virtual-environments

**Comment by JeremyLARDENOIS** [comment]
@AlexandreBrg which part do you want that i remove so ?

**Pull request closed without merging**